cli/src/main/kotlin/com/bazel_diff/hash/TargetHash.kt

```diff
@@ -2,17 +2,38 @@ package com.bazel_diff.hash
 
 data class TargetHash(
     val type: String, // Rule/GeneratedFile/SourceFile/...
-    val hash: String
+    val hash: String,
+    val directHash: String,
+    val deps: List<String>? = null
 ) {
     val hashWithType by lazy {
-        "${type}#${hash}"
+        "${type}#${hash}#${directHash}"
+    }
+
+    val totalHash by lazy {
+        "${hash}#${directHash}"
```

Comment:
Right now, I have `generate-hashes` always producing the overall and direct hash in the output file. I could put it behind a `--detailed-hashes` flag if desired.

Reply:
This change will be behind a breaking change, maybe the get-impacted-targets command can simply be aware of this migration

Reply:
oh, good point. I'll switch to using a different delimiter for hash and direct hash. That should cause it to work in a way if you use the new values with the old get-impacted-targets implementation. Old implementations will just see a longer hash string if they don't know how to split them out.